src/codegen/abi.js

```diff
@@ -496,83 +497,91 @@ module.exports = class AbiCodeGenerator {
+          input.get('name'),
+          input.get('type') === 'tuple'
+            ? this._tupleTypeName(input, index, tupleInputParentType, this.abi.name)
+            : util.isTupleArrayType(input.get('type'))
+            ? `Array<${this._tupleTypeName(input, index, tupleInputParentType)}>`
+            : typesCodegen.ascTypeForEthereum(input.get('type')),
+        ),
+      )
+
+      let superInputs = `
+      '${fnName}',
+      [${
+        inputs.size > 0
+          ? inputs
+              .map(input =>
+                typesCodegen.ethereumValueFromAsc(input.get('name'), input.get('type')),
+              )
+              .map(coercion => coercion.toString())
+              .join(', ')
+          : ''
+      }]`
+
+      let methodCallBody = isTry =>
+        `
+        let result = ${isTry ? 'super.tryCall' : 'super.call'}(${superInputs})
+        ${
+          isTry
+            ? `if (result.reverted) {
+            return new CallResult()
+          }
+          let value = result.value
           `
-          let result = super.call(
-            '${fnName}',
-            [${
-              inputs.size > 0
-                ? inputs
-                    .map(input =>
-                      typesCodegen.ethereumValueFromAsc(
-                        input.get('name'),
-                        input.get('type'),
-                      ),
-                    )
-                    .map(coercion => coercion.toString())
-                    .join(', ')
-                : ''
-            }]
-          );
-          return ${
-            simpleReturnType
-              ? typesCodegen.ethereumValueToAsc(
-                  'result[0]',
-                  outputs.get(0).get('type'),
-                  util.isTupleArrayType(outputs.get(0).get('type'))
-                    ? this._generateTupleType(
-                        outputs.get(0),
-                        0,
-                        tupleResultParentType,
-                        this.abi.name,
-                        '',
-                      ).name
-                    : '',
-                )
-              : `new ${returnType.name}(
-                  ${outputs
-                    .map(
-                      (output, index) =>
-                        `${typesCodegen.ethereumValueToAsc(
-                          `result[${index}]`,
-                          output.get('type'),
-                        )} ${
-                          output.get('type') === 'tuple'
-                            ? 'as ' +
-                              this._generateTupleType(
-                                output,
-                                index,
-                                tupleResultParentType,
-                                this.abi.name,
-                                '',
-                              ).name
-                            : ''
-                        }`,
+            : ''
+        }
+        return ${isTry ? 'CallResult.fromValue(' : ''} ${
```

Comment:
This section might be easier to read if there was just one top-level `isTry`-based conditional rather than a few spread across the nested logic.